src/drc/api/viewsets.py

```diff
@@ -208,6 +219,48 @@ def perform_destroy(self, instance):
 
         super().perform_destroy(instance.canonical)
 
+    def destroy_document_version(self, document: EnkelvoudigInformatieObject, versie: int) -> None:
+        document_version = EnkelvoudigInformatieObject.objects.filter(
+            identificatie=document.identificatie, bronorganisatie=document.bronorganisatie, versie=versie
+        )
+        if not document_version.exists():
+            raise serializers.ValidationError(_("The document version requested does not exist"),
+                code="wrong-version-number",
+            )
+
+        # Check if there are OIOs related to this specific version of the document
+        oios_related_to_version = ObjectInformatieObject.objects.filter(informatieobject=document.canonical, informatieobject_versie=versie)
+
+        if oios_related_to_version.exists():
+            raise serializers.ValidationError(
+                {
+                    api_settings.NON_FIELD_ERRORS_KEY: _(
+                        "All relations to the document version must be destroyed before destroying the document"
+                    )
+                }, code="pending-relations",
+            )
+
+        # Check if this document version is the last one remaining. If it is, check if there are related OIOs
+        # without version specified
+        documents_in_history = EnkelvoudigInformatieObject.objects.filter(
+            identificatie=document.identificatie, bronorganisatie=document.bronorganisatie
+        )
```

Comment:
you should filter on `uuid` and not on `identificatie` :) `(uuid, canonical)` are unique together

Reply:
I think here I meant to find all the versions of a document, so maybe this should have been a filter on `canonical`

Reply:
yeah, but even different versions of the same document have the same `uuid`. The end result should be the same for `uuid` or `canonical`.